### PR TITLE
8261395: C1 crash "cannot make java calls from the native compiler" 

### PR DESCRIPTION
**Pull request by dholmes-ora**
DRAFT

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Issue
 * [JDK-8261395](https://bugs.openjdk.java.net/browse/JDK-8261395): C1 crash "cannot make java calls from the native compiler"

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3895/head:pull/3895` \
`$ git checkout pull/3895`

Update a local copy of the PR: \
`$ git checkout pull/3895` \
`$ git pull https://git.openjdk.java.net/jdk pull/3895/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3895`

View PR using the GUI difftool: \
`$ git pr show -t 3895`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3895.diff">https://git.openjdk.java.net/jdk/pull/3895.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/oops/instanceKlass.cpp [hunk 1]
```diff
@@ -198,23 +196,41 @@ bool InstanceKlass::has_nest_member(InstanceKlass* k, TRAPS) const {
...
-          // can't have two names the same, so we're done
-          return false;
+        // Names match so check actual klass, as we need identity between classes not equivalence.
+        // It should not be possible for constant-pool resolution to trigger actual class-loading,
+        // as that would imply one classloader has defined two classes with the same name. But it
+        // is possible that the resolution process could require protection domain validation, which
+        // requires executing Java code, and can lead to exceptions (like OOME or StackOverflow).
+        // So we have to skip constant-pool resolution in that case.
+        if (can_resolve) {
+          Klass* k2 = _constants->klass_at(cp_index, THREAD);
```
Comment (coleenp) [inline src/hotspot/share/oops/instanceKlass.cpp:206]: why is this not just CHECK_NULL?

**Comment by coleenp** [comment]
And you should add Ioi's test.

File: src/hotspot/share/oops/instanceKlass.cpp (hunk 1 above)
Comment (dholmes-ora) [reply src/hotspot/share/oops/instanceKlass.cpp:206]: Logging. Plus the check for only VME which probably has to go anyway.

**Comment by dholmes-ora** [comment]
Scrapping this version and simplifying the overall process.

**Pull request closed without merging**